Authenticator/Source/SegmentedControlRowModel.swift

```diff
@@ -25,63 +25,46 @@
 
 import OneTimePassword
 
-protocol SegmentedControlRowModel {
-    typealias Action
+struct SegmentedControlRowModel<Action> {
+    let segments: [(title: String, action: Action)]
+    let selectedSegmentIndex: Int?
 
-    var segments: [(title: String, action: Action)] { get }
-    var selectedSegmentIndex: Int? { get }
+    init<V: Equatable>(options: [(title: String, value: V)], value: V, @noescape changeAction: (V) -> Action) {
```

Comment:
Line Length Violation: Line should be 100 characters or less: currently 111 characters (line_length)